com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/HDShaderPasses.cs

```diff
@@ -89,10 +94,35 @@ public static PassDescriptor GenerateScenePicking()
                 // Collections
                 renderStates = CoreRenderStates.ScenePicking,
                 pragmas = CorePragmas.DotsInstancedInV1AndV2EditorSync,
-                defines = CoreDefines.ScenePicking,
+                defines = GenerateDefines(),
                 includes = GenerateIncludes(),
+#if AddSkinDefineAndInclude
+                requiredFields = GenerateRequiredFields(),
+#endif
             };
 
+#if AddSkinDefineAndInclude
+            FieldCollection GenerateRequiredFields()
+            {
+                return new FieldCollection
+                {
+                    HDStructFields.AttributesMesh.vertexID
```

Comment:
Is there a better way to add vertexID, define and include? Not nice to add to every pass manually like this

Reply:
Included in shadder file instead, define removed, vertex ID always added for dots instancing